spec/system/candidate_interface/references/reference_selection/candidate_can_continuously_request_references_spec.rb

```diff
@@ -0,0 +1,195 @@
+require 'rails_helper'
+
+RSpec.feature 'References' do
+  include CandidateHelper
+
+  scenario 'the candidate can continue to request and add references on an unsubmitted application' do
+    given_i_am_signed_in
+    and_i_have_provided_my_personal_details
+    and_the_select_references_flag_is_active
+    and_i_have_three_reference_requests_pending
+
+    when_i_receive_two_references
+    then_i_still_have_a_reference_request_outstanding
+    and_i_can_add_more_reference_requests
+    and_i_can_receive_more_references
+  end
+
+  def given_i_am_signed_in
+    @candidate = create(:candidate)
+    login_as(@candidate)
+    @application = @candidate.current_application
+  end
+
+  def and_i_have_provided_my_personal_details
+    @candidate.current_application.update!(first_name: 'Mr', last_name: 'Bot')
+  end
+
+  def and_the_select_references_flag_is_active
+    FeatureFlag.activate(:reference_selection)
+  end
+
+  def and_i_have_three_reference_requests_pending
+    candidate_provides_three_referees
+  end
+
+  def when_i_receive_two_references
+    receive_references
+  end
+
+  def then_i_still_have_a_reference_request_outstanding
+    visit candidate_interface_references_review_path
+    expect(page).to have_content 'Awaiting response'
+  end
+
+  def and_i_can_add_more_reference_requests
+    when_i_visit_the_references_section
+    then_i_can_still_request_another_reference
+
+    when_i_see_the_type_page
+    and_i_select_academic
+    then_i_should_see_the_referee_name_page
+
+    when_i_fill_in_my_references_name
+    then_i_see_the_referee_email_page
+
+    when_i_provide_a_valid_email_address
+    then_i_see_the_relationship_page
+
+    when_i_fill_in_my_references_relationship
+    then_i_should_see_the_review_unsubmitted_page
+    and_i_should_see_my_references_details
+
+    when_i_choose_to_submit_my_reference_now
+    then_i_should_be_told_my_reference_request_has_been_sent
+  end
+
+  def and_i_can_receive_more_references
```

Comment:
I think we can just call the submit reference service here, since we're already testing all of this in a different system spec. 

So just 
```
reference = @application.application_references.last
SubmitReference.new(reference).save!

```